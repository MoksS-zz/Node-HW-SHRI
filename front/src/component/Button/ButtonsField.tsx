import React from "react";
import { withNaming } from '@bem-react/classname';

const cn = withNaming({ e: '_', m: '__', v: '-' })

const styles = cn('ButtonsField');

const ButtonsField = (props) => {
  let classes;

  if (props.style === undefined) {
    classes = "ButtonsField"
  } else {
    classes = styles(props.style);
  }

  return (
    <div className={classes}>
      {props.children}
    </div>
  )

};

export default ButtonsField;