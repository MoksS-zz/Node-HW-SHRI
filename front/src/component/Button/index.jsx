import React from "react";
import { withNaming } from '@bem-react/classname';
import "./Buttons.scss";

const cn = withNaming({ e: '_', m: '__', v: '-' })

const styles = cn('Button');

const Button = (props) => {
  let classes;

  if (props.style === undefined) {
    classes = "Button Button__color-control Button__padding-control Button__height-default"
  } else {
    classes = styles(props.style);
  }

  return (
    <button 
      className={classes}
      onClick={props.onClick}>
      {props.children}
    </button>
  )

};

export default Button;