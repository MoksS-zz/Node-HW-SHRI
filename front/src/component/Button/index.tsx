import React, { MouseEvent, FC } from "react";
import cn from "../../helpers/bemCn";
import "./Buttons.scss";

const styles = cn('Button');

export interface ButtonProps {
  style?: {
    color?: "close" | "action" | "control";
    padding?: "action" | "control";
    indentRigth?: "s";
    indentBottom?: "s" | "xl";
    height?: "action" | "default";
    indentLeft?: "s"
  };
  onClick?(event: MouseEvent<HTMLButtonElement>): void;
}

const Button: FC<ButtonProps> = (props) => {
  let classes: string;

  if (props.style === undefined) {
    classes = "Button Button__color-control Button__padding-control Button__height-default"
  } else {
    classes = styles(props.style);
  }

  return (
    <button 
      className={classes}
      onClick={props.onClick}>
      {props.children}
    </button>
  )

};

export default Button;