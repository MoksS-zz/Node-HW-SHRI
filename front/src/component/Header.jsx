import React from "react";
import { withNaming } from '@bem-react/classname';

const cn = withNaming({ e: '_', m: '__', v: '-' })

const Banner = cn('Header', 'Banner');

const Header = (props) => {
  let classes;

  if (props.class === undefined) {
    classes = "Header_Banner Header_Banner__color-default"
  } else {
    classes = Banner(props.class);
  }

  return (
    <header className="Header">
    <div className="Header_Content">
    <h1 className={classes}>{props.text}</h1>
    {props.children}
    </div>
  </header>
  )

};

export default Header;