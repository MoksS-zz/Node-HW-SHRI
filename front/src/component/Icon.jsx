import React from "react";
import { withNaming } from '@bem-react/classname';

const cn = withNaming({ e: '_', m: '__', v: '-' })

const styles = cn('Icon');

const Icon = (props) => {
  let classes;

  if (props.style === undefined) {
    classes = "Icon"
  } else {
    classes = styles(props.style);
  }

  return <span className={classes} />
};

export default Icon;