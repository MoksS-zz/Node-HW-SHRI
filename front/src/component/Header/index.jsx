import React from "react";
import { Link } from "react-router-dom"
import { withNaming } from '@bem-react/classname';
import "./Header.scss";

const cn = withNaming({ e: '_', m: '__', v: '-' })

const Banner = cn('Header', 'Banner');

const Header = (props) => {
  let classes;

  if (props.class === undefined) {
    classes = "Header_Banner Header_Banner__color-default"
  } else {
    classes = Banner(props.class);
  }

  return (
    <header className="Header">
    <div className="Header_Content">
    
    {props.link ? 
      <Link to={props.link}>
        <h1 className={classes}>{props.text}</h1>
      </Link>
      : <h1 className={classes}>{props.text}</h1>
    }
    {props.children}
    </div>
  </header>
  )

};

export default Header;