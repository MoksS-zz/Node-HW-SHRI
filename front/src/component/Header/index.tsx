import React, { FC } from "react";
import { Link } from "react-router-dom"
import cn from "../../helpers/bemCn";
import "./Header.scss";

const Banner = cn('Header', 'Banner');

export interface HeadProps {
  link?: string;
  text: string;
  class?: {
    color?: "black"
    lineHeight: "l"
  }; 
}

const Header: FC<HeadProps> = (props) => {
  let classes;

  if (props.class === undefined) {
    classes = "Header_Banner Header_Banner__color-default"
  } else {
    classes = Banner(props.class);
  }

  return (
    <header className="Header">
    <div className="Header_Content">
    
    {props.link ? 
      <Link to={props.link}>
        <h1 className={classes}>{props.text}</h1>
      </Link>
      : <h1 className={classes}>{props.text}</h1>
    }
    {props.children}
    </div>
  </header>
  )

};

export default Header;