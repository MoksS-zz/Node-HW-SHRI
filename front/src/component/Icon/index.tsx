import React, { FC } from "react";
import cn from "../../helpers/bemCn";
import "./Icon.scss";

const styles = cn('Icon');

export interface IconProps {
  style?: {
    size?: "xs" | "xl";
    img?: "close" | "control" | "play" | "rebuild";
  };
}

const Icon: FC<IconProps> = (props) => {
  let classes;

  if (props.style === undefined) {
    classes = "Icon"
  } else {
    classes = styles(props.style);
  }

  return <span className={classes} />
};

export default Icon;