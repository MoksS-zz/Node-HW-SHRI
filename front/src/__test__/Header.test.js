import React from "react";
import renderer from "react-test-renderer";
import Header from "../component/Header";
import Button from "../component/Button";
import ButtonsField from "../component/Button/ButtonsField";
import Icon from "../component/Icon";
import Text from "../component/Text";

const Link = (props) => <a href={props.to}>{props.children}</a>

describe("Header", () => {
  it("renders Header in Start Screen correctly", () => {
    const header = renderer
      .create(
        <Header text="School CI server">
          <Link to="/settings">
            <Button style={{ color: "control", padding: "control", height: "default" }}>
              <Icon style={{ size: "xl", img: "control" }} />
              <Text style={{ size: "m", lineHeight: "xxl", weight: "small", color: "default", hide: "on" }}>Settings</Text>
            </Button>
          </Link>
        </Header>
      )
      .toJSON();

    expect(header).toMatchSnapshot();
  });

  it("renders Header in Settings correctly", () => {
    const header = renderer
      .create(<Header text="School CI server" />)
      .toJSON();

    expect(header).toMatchSnapshot();
  });

  it("renders Header in Build correctly", () => {
    const header = renderer
      .create(
        <Header
          text="MoksS/JESTTEST"
          class={{ color: "black", lineHeight: "l" }}
        >

          <ButtonsField style={{ column: "off" }}>
            <Button
              style={{ color: "control", padding: "control", height: "default" }}
            >
              <Icon style={{ size: "xl", img: "play" }} />
              <Text style={{ size: "m", lineHeight: "xxl", weight: "small", color: "default", hide: "on" }}>Run build</Text>
            </Button>

            <Link to="/settings">
              <Button style={{ color: "control", indentLeft: "s", height: "default" }}>
                <Icon style={{ size: "xl", img: "control" }} />
              </Button>
            </Link>
          </ButtonsField>
        </Header>
      )
      .toJSON();

    expect(header).toMatchSnapshot();
  });
});


