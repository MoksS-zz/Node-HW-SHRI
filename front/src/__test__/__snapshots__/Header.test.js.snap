// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header in Build correctly 1`] = `
<header
  className="Header"
>
  <div
    className="Header_Content"
  >
    <h1
      className="Header_Banner Header_Banner__color-black Header_Banner__lineHeight-l"
    >
      MoksS/JESTTEST
    </h1>
    <div
      className="ButtonsField ButtonsField__column-off"
    >
      <button
        className="Button Button__color-control Button__padding-control Button__height-default"
      >
        <span
          className="Icon Icon__size-xl Icon__img-play"
        />
        <span
          className="Text Text__size-m Text__lineHeight-xxl Text__weight-small Text__color-default Text__hide-on"
        >
          Run build
        </span>
      </button>
      <a
        href="/settings"
      >
        <button
          className="Button Button__color-control Button__indentLeft-s Button__height-default"
        >
          <span
            className="Icon Icon__size-xl Icon__img-control"
          />
        </button>
      </a>
    </div>
  </div>
</header>
`;

exports[`Header renders Header in Settings correctly 1`] = `
<header
  className="Header"
>
  <div
    className="Header_Content"
  >
    <h1
      className="Header_Banner Header_Banner__color-default"
    >
      School CI server
    </h1>
  </div>
</header>
`;

exports[`Header renders Header in Start Screen correctly 1`] = `
<header
  className="Header"
>
  <div
    className="Header_Content"
  >
    <h1
      className="Header_Banner Header_Banner__color-default"
    >
      School CI server
    </h1>
    <a
      href="/settings"
    >
      <button
        className="Button Button__color-control Button__padding-control Button__height-default"
      >
        <span
          className="Icon Icon__size-xl Icon__img-control"
        />
        <span
          className="Text Text__size-m Text__lineHeight-xxl Text__weight-small Text__color-default Text__hide-on"
        >
          Settings
        </span>
      </button>
    </a>
  </div>
</header>
`;
